### PR TITLE
Make dataset preview for h5mlm datatype

### PR DESCRIPTION
**Pull request by qiagu**
## What did you do? 
- [describe the proposed changes]
- make dataset preview for h5mlm datatype

## Why did you make this change?
(Cite Issue number OR provide rationalization of changes if no issue exists)
(If fixing a bug, please add any relevant error or traceback)

## How to test the changes? 
(select the most appropriate option; if the latter, provide steps for testing below)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

## For UI Components
- [x] I've included a screenshot of the changes

### CODE REVIEW

**Comment by qiagu** [comment]
image

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -1512,11 +1515,12 @@ def get_config_string(self, filename):
             return config
         except Exception as e:
             log.warning('%s, get model configuration Except: %s', self, e)
-            return ""
+            return "{}"
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]: Is this a string representation of an empty dictionary? If so, why not just leave the `log.warning` line and let the caller handle `None`? 
~~Also, out of curiosity, where is this method called from?~~ (I see now; was looking at an older branch)

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -1528,6 +1532,35 @@ def display_peek(self, dataset):
...
+
+        if to_ext or not preview:
+            to_ext = to_ext or dataset.extension
+            return self._serve_raw(trans, dataset, to_ext, **kwd)
+
+        rval = {}
+        try:
+            with h5py.File(dataset.file_name, "r") as handle:
+                rval['Attributes'] = {}
+                for k in handle.attrs.keys():
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]: I think you don't need `keys()` here: `h5py.AttributeManager` supports a dict interface, and by doing `for k in handle.attrs` you'll be iterating over the keys (see the `__iter__` method in the [docs](https://docs.h5py.org/en/latest/high/attr.html#reference))

EDIT: However, if instead you do this: `for k, v in handle.attrs.items():` you get the `v` for free and you don't have to do the extra lookup on line 1549.

File: lib/galaxy/datatypes/binary.py [hunk 3]
```diff
@@ -1528,6 +1532,35 @@ def display_peek(self, dataset):
...
+                    if k in ('-URL-', '-repr-'):
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]: or negate it, and then you don't need `continue`, so less code to read. OK, either way.

File: lib/galaxy/datatypes/binary.py [hunk 4]
```diff
@@ -1528,6 +1532,35 @@ def display_peek(self, dataset):
...
+                        continue
+                    v = util.unicodify(handle.attrs.get(k))
+                    rval['Attributes'][k] = v
+        except Exception as e:
+            log.warning(e)
+            pass
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]:
```suggestion
```

File: lib/galaxy/datatypes/binary.py [hunk 5]
```diff
@@ -1528,6 +1532,35 @@ def display_peek(self, dataset):
...
+
+        config = self.get_config_string(dataset.file_name)
+        rval['Config'] = json.loads(config)
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]: I think this should be conditional on the value of `config`. And `get_config_string` doesn't have to return `"{}"`: if you need it to be a str repr of a dict even if it's empty, you can do this:
```py
rval['Config'] = json.loads(config) if config else '{}'
```

**Comment by ic4f** [comment]
@qiagu thank you for addressing the comments! Everything looks good!

**Review by jdavcs — APPROVED** [review]

**Pull request merged**